src/app/features/data-import/data-import/data-import.component.ts

```diff
@@ -58,34 +56,31 @@ export class DataImportComponent {
     public entities: EntityRegistry
   ) {}
 
-  // TODO add supported types for file select
-  async setCsvFile(inputEvent: Event): Promise<void> {
-    const file = this.getSelectedFile(inputEvent);
-    try {
-      this.csvFile = await this.loadCSVFile(file);
-      this.fileNameForm.setValue({ fileName: file.name });
-      this.columnMappingForm = new FormRecord<FormControl<string>>({});
-      this.csvFile.meta.fields.forEach((field) =>
-        this.columnMappingForm.addControl(field, new FormControl())
-      );
-      this.entitySelectionChanged();
-    } catch (e) {
-      this.fileNameForm.setErrors({ fileInvalid: e.message });
-    }
-    this.changeDetectorRef.detectChanges();
-  }
-
-  private getSelectedFile(inputEvent: Event) {
-    const target = inputEvent.target as HTMLInputElement;
-    return target.files[0];
-  }
+  async loadData(parsedData: ParsedData): Promise<void> {
+    this.importData = parsedData;
 
-  private async loadCSVFile(file: File) {
     this.entityForm.enable();
     this.transactionIDForm.enable();
-    const csvFile = await this.dataImportService.validateCsvFile(file);
-    if (csvFile.meta.fields.includes("_id") && csvFile.data[0]["_id"]) {
-      const record = csvFile.data[0] as { _id: string };
+
+    this.updateConfigFromDataIds();
+
+    this.updateColumnMappingFromData();
+
+    this.entitySelectionChanged();
+
+    this.readyForImport = true;
+  }
+
+  /**
+   * Check if new data contains _id and infer config options.
+   * @private
+   */
+  private updateConfigFromDataIds() {
+    if (
+      this.importData.meta.fields.includes("_id") &&
+      this.importData.data[0]["_id"]
+    ) {
+      const record = this.importData.data[0] as { _id: string };
       if (record._id.includes(":")) {
```

Comment:
```suggestion
      if (record._id.toString().includes(":")) {
```
When I just entered a number as id, this failed because it was parsed as a number and not a string.